Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -10455,7 +10458,23 @@ void WebPageProxy::gamepadActivity(const Vector<std::optional<GamepadData>>& gam
     send(Messages::WebPage::GamepadActivity(gamepadDatas, eventVisibility));
 }
 
-#endif
+void WebPageProxy::recentGamepadAccessStateChanged(PAL::HysteresisState state)
+{
+    switch (state) {
+    case PAL::HysteresisState::Started:
+        m_uiClient->recentlyAccessedGamepadsForTesting(*this);
+        break;
+    case PAL::HysteresisState::Stopped:
+        m_uiClient->stoppedAccessingGamepadsForTesting(*this);
+    }
+}
+
+void WebPageProxy::gamepadsRecentlyAccessed()
```

Comment:
`WebPageProxy` should validate the message sender has access to game pad; maybe add a FIXME as a reminder if we are not going to address it here.